aiida/backends/sqlalchemy/migrations/env.py

```diff
@@ -30,20 +30,19 @@ def run_migrations_online():
     from aiida.backends.sqlalchemy.models.base import Base
     config = context.config  # pylint: disable=no-member
 
-    connectable = config.attributes.get('connection', None)
+    connection = config.attributes.get('connection', None)
 
-    if connectable is None:
+    if connection is None:
         from aiida.common.exceptions import ConfigurationError
         raise ConfigurationError('An initialized connection is expected for the AiiDA online migrations.')
 
-    with connectable.connect() as connection:
-        context.configure(  # pylint: disable=no-member
-            connection=connection,
-            target_metadata=Base.metadata,
-            transaction_per_migration=True,
-        )
+    context.configure(  # pylint: disable=no-member
+        connection=connection,
+        target_metadata=Base.metadata,
+        transaction_per_migration=True,
+    )
```

Comment:
What happened here, how did this stop needing a context? There is no longer any tear down to do after you are done with the connection? (and if not, what was there before that is no longer necessary?)

Reply:
Well I personally re-wrote the documentation on it in alembic, so I can tell you it is definitely correct 😉 https://github.com/sqlalchemy/alembic/pull/911 (see also the discussion in the related issue)

Reply:
So from [this](https://github.com/sqlalchemy/alembic/issues/908#issuecomment-916146335) I understand that apparently the Alembic library is supposed to manage the connection "behind the curtain" and they take care of opening the context during the migration process?

Reply:
yep exacty: the connection has already been opened, so we don't need to